SubscriptionManager/src/test/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionServiceTest.java

```diff
@@ -1,146 +1,339 @@
+
+	@Mock
+	IntervalNotificationHandler intervalHandlerREST;
+
+	@Mock
+	InternalNotification notify;
+
+	@Mock
+	HashMap<String, Subscription> subscriptionId2Subscription;
+
+	@Inject
+	SubscriptionService subscriptionService;
+
+	@Mock
+	NotificationParam param;
+
+	@Mock
+	Subscription subscriptionMock;
+
+	@Mock
+	IntervalNotificationHandler intervalNotificationHandler;
+
+	@Inject
+	SubscriptionInfoDAO subService;
+
+	ArrayListMultimap<String, String> multimaparr = ArrayListMultimap.create();
+
+	// resolved payload of a subscription.
+
+	String payload = "{\r\n  \"https://uri.etsi.org/ngsi-ld/entities\" : [ "
+			+ "{\r\n    \"@type\" : [ \"https://uri.etsi.org/ngsi-ld/default-context/Vehicle\" ]"
+			+ "\r\n  } ],\r\n  \"@id\" : \"urn:ngsi-ld:Subscription:173223\","
+			+ "\r\n  \"https://uri.etsi.org/ngsi-ld/notification\" : [ "
+			+ "{\r\n    \"https://uri.etsi.org/ngsi-ld/endpoint\" : [ "
+			+ "{\r\n      \"https://uri.etsi.org/ngsi-ld/accept\" : [ "
+			+ "{\r\n        \"@value\" : \"application/json\""
+			+ "\r\n      } ],\r\n      \"https://uri.etsi.org/ngsi-ld/uri\" : [ {"
+			+ "\r\n        \"@value\" : \"http://localhost:8080/acc\"\r\n      } ]"
+			+ "\r\n    } ]\r\n  } ],\r\n  \"@type\" : [ \"https://uri.etsi.org/ngsi-ld/Subscription\" ]" + "\r\n}";
+
+	/**
+	 * this method is used to test create subscription test
+	 */
+	@Test
+	@Order(1)
+	public void serviceTest() throws ResponseException {
+
+		List<Object> context = new ArrayList<>();
+		Context context1 = new Context();
+		Subscription subscription = null;
+		
+		
+		ArrayListMultimap<String, String> subIds = ArrayListMultimap.create();
+		multimaparr.put("content-type", "application/json");
+		Gson gson = new Gson();
+		
+		Map<String, Object> resolved = gson.fromJson(payload, Map.class);
+		Subscription s = Subscription.expandSubscription(resolved, context1, true);
+		subscription = new Subscription();
+		
+		SubscriptionRequest subRequest = new SubscriptionRequest(s, context, subIds, 0);
+		String subId = subscriptionService.subscribe(subRequest).await().indefinitely();
+		
+		Assertions.assertEquals("urn:ngsi-ld:Subscription:173223", subId);
+		
+	}
+
+	/**
+	 * this method is used to test update subscription test
+	 */
+	@Test
+	@Order(2)
+	public void updateSubscriptionTest() {
+
+		try {
+			List<Object> context = new ArrayList<>();
+			Context context1 = new Context();
+			Subscription subscription = null;
+			
+			
+			ArrayListMultimap<String, String> subIds = ArrayListMultimap.create();
+			multimaparr.put("content-type", "application/json");
+			Gson gson = new Gson();
+			
+			Map<String, Object> resolved = gson.fromJson(payload, Map.class);
+			Subscription s = Subscription.expandSubscription(resolved, context1, true);
+			subscription = new Subscription();
+			SubscriptionRequest subRequest = new SubscriptionRequest(s, context, subIds, 0);
+			
+			subscriptionService.updateSubscription(subRequest).await().indefinitely();
```

Comment:
Add verify and compare expected and actual result.

Reply:
Done